### ISSUE TITLE
[Bug] Wrong number of items per row when rotating CollectionView

### ISSUE THREAD

**Issue by aliriomendes** [xamarin/Xamarin.Forms#7897]
### Description

I'm trying to have a dynamic number of item per row depending on the screen width.  
When I rotate the device and recalcula the number of items per row (GridItemsLayout.Span), the math is right but collection view doesn't behave accordantly on iOS. 

```
private void CollectionView_SizeChanged(object sender, EventArgs e)
{
    if (sender is CollectionView collectionView && collectionView.ItemsLayout is GridItemsLayout gridItemsLayout)
    {
        gridItemsLayout.Span = Math.Max(1, (int)Math.Floor(collectionView.Width / 320));
        System.Diagnostics.Debug.WriteLine("Items per row -> " + gridItemsLayout.Span);
    }
}
```

### Steps to Reproduce

1. Rotate device

### Basic Information

- Version with issue:
Any version of Xamarin.Forms

- Platform Target Frameworks: 
  - iOS:  

- Affected Devices:
iPad - Number of items in screen is different from  GridItemsLayout.Span
iPhone - Number of items in screen is different from  GridItemsLayout.Span when rotate to Horizontal and and item desapear when rotate to Vertical.
### Screenshots
![Simulator Screen Shot - iPad Air (3rd generation) - 2019-10-09 at 14 12 35](https://user-images.githubusercontent.com/6729864/66480515-2a300300-ea9f-11e9-8e18-f466598f662b.png)
![Simulator Screen Shot - iPad Air (3rd generation) - 2019-10-09 at 14 13 03](https://user-images.githubusercontent.com/6729864/66480524-2ef4b700-ea9f-11e9-89d0-4c5c51393508.png)

### Reproduction Link
[FormsCollectionViewRotationTest.zip](https://github.com/xamarin/Xamarin.Forms/files/3707277/FormsCollectionViewRotationTest.zip)

**Comment by aliriomendes** [comment]
I found a quick workaround:
```
if (sender is CollectionView collectionView && collectionView.ItemsLayout is GridItemsLayout gridItemsLayout)
{
    if (Device.RuntimePlatform == Device.iOS)
        await Task.Yield();
    gridItemsLayout.Span = Math.Max(1, (int)Math.Floor(collectionView.Width / 320));
}
```

**Comment by aliriomendes** [comment]
> I see some strange things going on. When rotating back the number of items is different than before on iPad and on iPhone (11) whenever I rotate back the screen just goes blank. This latter case might be due to new code changes that aren't in the NuGet yet.
> 
> Enough to investigate this further!
> 
> @aliriomendes did you happen to test this also on Android?

Yes, I tested it, and it works fine on Android.

**Comment by Jared-Aus** [comment]
I'm finding the whole GridItemsLayout class to be very static. In my testing so far (iOS only), the HorizontalItemSpacing, VerticalItemSpacing, and Span properties don't work with DynamicResource bindings. I also couldn't get the Span property to bind to my ViewModel.

**Comment by hs79hs** [comment]
Does anybody have a solution for this? I also found this issue. When rotate back, if the span is set to 2 or 3, it works like the span is 1. And if I set the span to 4, it shows like the span is 2

**Comment by ryl** [comment]
I'm encountering this problem on an iPhoneXr with iOS 13.3.1

**Comment by borgdylan** [comment]
any news on this?

**Comment by hartez** [comment]
This is fixed in 4.6.0.847.
